.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,33 @@
+---
+name: golangci-lint
+on:
+  push:
+    paths:
+      - "go.sum"
+      - "go.mod"
+      - "**.go"
+      - ".github/workflows/golangci-lint.yml"
+      - ".golangci.yml"
+  pull_request:
+
+permissions:  # added using https://github.com/step-security/secure-repo
+  contents: read
+
+jobs:
+  golangci:
+    permissions:
+      contents: read  # for actions/checkout to fetch code
+      pull-requests: read  # for golangci/golangci-lint-action to fetch pull requests
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      - name: install Go
+        uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5.0.0
+        with:
+          go-version: 1.22.x
```

Comment:
We use

```
module github.com/reddit/monoceros

go 1.19
```

should we match and update after?

Reply:
The `go` directive in Go modules is a minimum version specifier for compiler compatibility.

We could setup a matrix job to test all versions, but I've not found that necessary for running golangci-lint.

For `go test`, we'll want to setup a compatibility test matrix.

Reply:
TIL, makes sense